Charge_pi/src/BaseStationMotorTest.cpp

```diff
@@ -0,0 +1,84 @@
+#include <wiringPi.h>
+#include <stdio.h>
+#include <iostream>
+#include "HX711.h"
+
+int pin1 = 2;
+int pin2 = 3;
+int pin4 = 5;
+int motorPin1 = 8;
+int motorPin2 = 9;
+
+HX711 scale;
+
+uint8_t dataPin = 0;
+uint8_t clockPin = 1;
+
+uint32_t offset = 66483;
+float set_Scale = -140.714996;
+
+float volatile f;
+
+
+int main() {
+  // put your setup code here, to run once:
+  wiringPiSetup();
+  pinMode(pin1, INPUT);
+  pinMode(pin2, INPUT);
+  pinMode(pin4, INPUT);
+
+  pinMode(motorPin1, OUTPUT);
+  pinMode(motorPin2, OUTPUT);
+  
+  scale.begin(dataPin, clockPin);
+  scale.set_scale(set_Scale);
+  scale.set_offset(offset);
+
+while(1) {
+  f = scale.get_units(5);
+  std::cout << "load:" << f << std::endl;
+  delay(250);
+
+  if(!digitalRead(pin4)){
+      digitalWrite(motorPin1, LOW);
+      digitalWrite(motorPin2, LOW);
+      f = scale.get_units(10);
+      std::cout << "pin 4 goed \n";
+      if(f < 2000){
+        std::cout << "load cell goed \n";
+        if(!digitalRead(pin1)){
+          std::cout << "pin 1 goed, arm up\n";
+            //Arm goes up.
+          digitalWrite(motorPin1, HIGH);
+          digitalWrite(motorPin2, LOW);
+        }
+        if(!digitalRead(pin2)){
+          std::cout << "pin 2 goed, arm still \n";
+            // Arm is max up.
+          digitalWrite(motorPin1, LOW);
+          digitalWrite(motorPin2, LOW);
+
+        }
+      }else{
```

Comment:
Stroustrup K&R

Reply:
This code doesn't matter anymore. 
full version in main.cpp